Make code checks pass

diegoceccarelli opened the pull request:
# Fix linting issues and typos across the codebase

## Description
This PR addresses various linting issues and typos identified by the linters (ruff, black, blackdoc, and typos). The changes include:
- Removing `>>>` prompts from code blocks in markdown files
- Fixing docstring examples in Python files
- Correcting typos (e.g., "relevants" to "relevant")
- Updating the ruff configuration to use the newer format
- Replacing abbreviated journal name with full name in bibliographic references

## Changes
### Configuration
- Updated `pyproject.toml` to use the newer ruff configuration format
- Removed unused typos configuration

### Documentation
- Fixed code blocks in `README.md` and `docs/index.md` by removing `>>>` prompts
- Updated journal name from `Comput. Medical Imaging Graph.` to `Computerized Medical Imaging and Graphics` in:
  - `ranx/fusion/isr.py`
  - `ranx/fusion/logn_isr.py`
  - `ranx/fusion/log_isr.py`
  - `docs/fusion.md`

### Code
- Fixed docstring examples in `ranx/data_structures/report.py`
- Corrected typo "relevants" to "relevant" in:
  - `changelog.md`
  - `ranx/metrics/interpolated_precision_at_recall.py`

## Testing
- Ran `make lint` to verify that all linting checks pass:
  - isort
  - black
  - blackdoc
  - ruff
  - typos

AmenRa commented on docs/fusion.md:
```diff
@@ -296,7 +296,7 @@ Computes ISR as proposed by [Mourão et al.](https://www.sciencedirect.com/scien
                     Jo{\~{a}}o Magalh{\~{a}}es},
         title     = {Multimodal medical information retrieval with unsupervised rank fusion},
-        journal   = {Comput. Medical Imaging Graph.},
+        journal   = {Computerized Medical Imaging and Graphics},
```
Ok.

AmenRa commented on docs/fusion.md:
```diff
@@ -344,7 +344,7 @@ Computes Log_ISR as proposed by [Mourão et al.](https://www.sciencedirect.com/s
                     Jo{\~{a}}o Magalh{\~{a}}es},
         title     = {Multimodal medical information retrieval with unsupervised rank fusion},
-        journal   = {Comput. Medical Imaging Graph.},
+        journal   = {Computerized Medical Imaging and Graphics},
```
Ok.

AmenRa commented on ranx/fusion/isr.py:
```diff
@@ -45,7 +45,7 @@ def isr(runs: List[Run], name: str = "isr") -> Run:
                         Jo{\~{a}}o Magalh{\~{a}}es},
             title     = {Multimodal medical information retrieval with unsupervised rank fusion},
-            journal   = {Comput. Medical Imaging Graph.},
+            journal   = {Computerized Medical Imaging and Graphics},
```
Ok.

AmenRa commented on ranx/fusion/log_isr.py:
```diff
@@ -51,7 +51,7 @@ def log_isr(runs: List[Run], name: str = "log_isr") -> Run:
                         Jo{\~{a}}o Magalh{\~{a}}es},
             title     = {Multimodal medical information retrieval with unsupervised rank fusion},
-            journal   = {Comput. Medical Imaging Graph.},
+            journal   = {Computerized Medical Imaging and Graphics},
```
Ok.

AmenRa commented on ranx/fusion/logn_isr.py:
```diff
@@ -51,7 +51,7 @@ def logn_isr(runs: List[Run], sigma: float = 0.01, name: str = "logn_isr") -> Ru
                         Jo{\~{a}}o Magalh{\~{a}}es},
             title     = {Multimodal medical information retrieval with unsupervised rank fusion},
-            journal   = {Comput. Medical Imaging Graph.},
+            journal   = {Computerized Medical Imaging and Graphics},
```
Ok.

AmenRa commented on ranx/meta/evaluate.py:
Ok.

AmenRa commented:
I commented the changes to keep.

I think we should do the following.

#### Make `pyproject.toml` as follows:
```
[tool.ruff.lint]
# E501 ignore long comment lines (black should take care of that)
# E712 ignore == False, needed for numba checks.
ignore = ["E501", "E712"]

[tool.isort]
profile = "black"
```

#### Make `_typos.toml` as follows to ignore "relevants":
```
[default]
extend-ignore-identifiers-re = [
    ".*Hsi",
]

[default.extend-words]
# Prevent Typos from flagging these as typos
relevants = "relevants"
```

#### Remove `$(VENV_BIN)/blackdoc .` from `Makefile lint`.
Docs formatting is fine as-is, in my opinion.

diegoceccarelli commented:
## PR Updated - Reviewer Comments Addressed

  I've updated this PR to address all the feedback. The branch has been rebased on master with a single clean commit.

  ### Changes Made

  **🔧 Tool Configuration Fixes:**
  - Configured Black to exclude markdown files via `pyproject.toml` to prevent formatting conflicts
  - Removed blackdoc from linting pipeline (no `.rst` files in project, providing no value)
  - Added IR-specific terminology to typos ignore list (`"relevants"`, `"Comput"`)

  **📝 Documentation Formatting Restored:**
  - Reverted to reviewer-preferred compact dictionary formatting in README and docs
  - Restored aligned inline comments and `>>>` output patterns
  - Maintained IR domain terminology (`"relevants"` instead of `"relevant"`)
  - Fixed syntax errors in documentation examples

  **🐛 Technical Fixes:**
  - Fixed syntax error in `report.py` docstring by changing code block type from `python` to plain text
  - Preserved all technical improvements (bug fixes, type checking, tool updates)

  ### Verification

  The updated configuration ensures:
  - ✅ `make lint` passes cleanly
  - ✅ `make fix-lint` respects markdown file exclusions
  - ✅ No formatting conflicts between tools and human preferences
  - ✅ All technical improvements preserved

  The linting pipeline is now streamlined to: `isort` → `black` → `ruff` → `typos`
